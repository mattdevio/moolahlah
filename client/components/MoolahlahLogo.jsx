/*----------  Vendor Imports  ----------*/
import styled from 'styled-components';

/**
 * Moolahlah Logo
 * The bread & butter of the brand.
 */
// const MoolahlahLogo = styled.span`
//   background: url(/assets/image/moolahlah_logo.png) no-repeat;
//   background-size: 30rem 8.9rem;
//   width: 30rem;
//   height: 8.9rem;
//   margin-bottom: 2rem;
//   display: inline-block;
// `;

const MoolahlahLogo = styled.span`
  background: url(/assets/image/moolahlah_logo.png) no-repeat;
  background-size: ${({ width }) => `${width}rem ${(+width/3.37)}rem`};
  width: ${({ width }) => `${width}rem`};
  height: ${({ width }) => `${+width/3.37}rem`};
  margin: ${({ margin }) => margin};
  display: block;
`;

MoolahlahLogo.defaultProps = {
  width: '30',
  margin: '0 0 2rem 0',
};

export default MoolahlahLogo;
