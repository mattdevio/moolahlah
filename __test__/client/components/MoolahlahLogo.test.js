// Vendor Imports
import React from 'react';
import { mount } from 'enzyme';

// Custom Imports
import MoolahlahLogo from '@/components/MoolahlahLogo';

describe('<MoolahlahLogo /> Component', () => {

  it('Should match the snapshot', () => {
    const snap = mount(<MoolahlahLogo />);
    expect(snap).toMatchSnapshot();
  });

});