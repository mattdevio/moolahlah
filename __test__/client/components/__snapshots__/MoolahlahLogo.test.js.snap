// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoolahlahLogo /> Component Should match the snapshot 1`] = `
.c0 {
  background: url(/assets/image/moolahlah_logo.png) no-repeat;
  background-size: 30rem 8.90207715133531rem;
  width: 30rem;
  height: 8.90207715133531rem;
  margin: 0 0 2rem 0;
  display: block;
}

<styled.span
  margin="0 0 2rem 0"
  width="30"
>
  <span
    className="c0"
    width="30"
  />
</styled.span>
`;
